esmf_regrid/tests/unit/experimental/io/test_round_tripping.py

```diff
@@ -248,11 +264,10 @@ def test_MeshToGridESMFRegridder_bilinear_round_trip(tmp_path):
     assert np.array_equal(original_matrix.todense(), loaded_matrix.todense())
 
     # Demonstrate regridding still gives the same results.
-    src_data = np.arange(np.product(src.data.shape)).reshape(src.data.shape)
-    src_mask = np.zeros(src.data.shape)
-    src_mask[0] = 1
-    src.data = ma.array(src_data, mask=src_mask)
+    src_data = ma.arange(np.product(src.data.shape)).reshape(src.data.shape)
+    src_data[0] = ma.masked
     assert np.array_equal(original_rg(src).data, loaded_rg(src).data)
+    assert np.array_equal(original_rg(src).data.mask, loaded_rg(src).data.mask)
```

Comment:
It looks like you're doing this full calculation again, rather than just doing it then checking the data and mask.

It also looks like you're not using `src_data` having made it, you're just sending `src` into the regridder without updating its data first?